### PR TITLE
Re-Authorize Tokens Integration Entity

### PR DESCRIPTION
**Pull request by DmitryTsymbal**
**Description**
This PR with a test that Re-Authorizes Tokens for created integration entity

 **Preconditions:**
 1. Create Integration.
 2. Activate Integration.
 
 **Steps:**
 1. Go to Integration page on backend.
 2. Click on the "Reauthorize" link on the Integration grid.
 3. Click on the "Reauthorize" button.
 4. Click Done.
 5. Perform assertions.

**Fixed Issues**
#519 Convert ReAuthorizeTokensIntegrationEntityTest to MFTF

### CODE REVIEW

File: app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            <grabValueFrom selector="{{AdminNewIntegrationSection.accessTokenSecret}}" stepKey="grabAccessTokenSecret"/>
+            <actionGroup ref="AdminAllowResourcesAccessIntegrationActionGroup" stepKey="finishTheProcess"/>
+            <actionGroup ref="AssertMessageInAdminPanelActionGroup" stepKey="seeSuccessMessage">
+                <argument name="message" value="The integration '{{defaultIntegrationData.name}}' has been re-authorized."/>
+            </actionGroup>
+            <actionGroup ref="AdminSearchIntegrationInGridActionGroup" stepKey="findReAuthorizedIntegration">
+                <argument name="name" value="{{defaultIntegrationData.name}}"/>
+            </actionGroup>
+            <actionGroup ref="AdminIntegrationOpenExistingEntityActionGroup" stepKey="openIntegrationEntity"/>
+            <actionGroup ref="AssertAdminIntegrationTokensAfterReauthoriseActionGroup" stepKey="assertTokens">
```
Comment (rogyar) [inline app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml]: We may improve the assertion a little bit. Please, take a look at the assertion in the original test. There's a call to [checkTokens](https://github.com/magento/magento2/blob/d1fb65c0f76947a78e28206a057d6597e4e32004/dev/tests/functional/tests/app/Magento/Integration/Test/Constraint/AssertIntegrationTokensAfterReauthorize.php#L62) method that ensures the changes are applied correctly. Applied correctly means that the consumer key and consumer secret must stay unchanged but the access token and the access toke secret should be regenerated. Basically, we should compare the initial values with the values after the token regeneration.

**Comment by rogyar** [comment]
By the way. Must admit that the level of your new PRs is much-much higher in comparison to the very beginning. Awesome job! Keep on keeping on 👍

**Comment by DmitryTsymbal** [comment]
Hello @rogyar , thank you for review and for the feedback, it's very pleasant.
I have refactored assertion, adding <assertEquals> and <assertNotEquals> for tokens comparing. Could you please review is this a correct way?
Thank you, regards!

File: app/code/Magento/Integration/Test/Mftf/ActionGroup/AdminAllowResourcesAccessIntegrationActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+
```
Comment (lbajsarowicz) [inline app/code/Magento/Integration/Test/Mftf/ActionGroup/AdminAllowResourcesAccessIntegrationActionGroup.xml]: Redundant space.

File: app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            <features value="Integration"/>
+            <stories value="System Integration"/>
+            <title value="Reauthorise Integration's Tokens"/>
+            <description value="ReAuthorising Tokens For Created Integration"/>
+            <group value="integration"/>
+            <group value="mtf_migrated"/>
+        </annotations>
+        <before>
+            <!-- Login As Admin -->
+            <actionGroup ref="LoginAsAdmin" stepKey="LoginAsAdmin"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml]: `stepKey` should be `camelCase` but Action Group name should be `PascalCase`.
Use `AdminLoginActionGroup`

File: app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        <after>
+            <actionGroup ref="AdminNavigateMenuActionGroup" stepKey="returnToIntegrationsPage">
+                <argument name="menuUiId" value="{{AdminMenuSystem.dataUiId}}"/>
+                <argument name="submenuUiId" value="{{AdminMenuSystemExtensionsIntegrations.dataUiId}}"/>
+            </actionGroup>
+            <actionGroup ref="AdminSearchIntegrationInGridActionGroup" stepKey="searchForReAuthorizedIntegration">
+                <argument name="name" value="{{defaultIntegrationData.name}}"/>
+            </actionGroup>
+            <actionGroup ref="AdminDeleteIntegrationEntityActionGroup" stepKey="deleteCreatedIntegration"/>
+            <actionGroup ref="logout" stepKey="logout"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml]: `AdminLogoutActionGroup`

File: app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        </after>
+            <!-- TEST BODY -->
```
Comment (lbajsarowicz) [inline app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml]: Avoid redundant comments that show up as a steps on Test Reports.

File: app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            </actionGroup>
+            <actionGroup ref="AdminSearchIntegrationInGridActionGroup" stepKey="findReAuthorizedIntegration">
+                <argument name="name" value="{{defaultIntegrationData.name}}"/>
+            </actionGroup>
+            <actionGroup ref="AdminIntegrationOpenExistingEntityActionGroup" stepKey="openIntegrationEntity"/>
+            <!-- Compare Tokens After Reauthorizing -->
+            <assertEquals stepKey="assertConsumerKey" expected="($grabConsumerKey)" actual="$grabAfterReauthorizeConsumerKey" actualType="string"/>
+            <assertEquals stepKey="assertConsumerSecret" expected="($grabConsumerSecret)" actual="$grabAfterReauthorizeConsumerSecret" actualType="string"/>
+            <assertNotEquals stepKey="assertNotEqualsToken" expected="($grabAccessToken)" actual="$grabReauthorizedAccessToken" actualType="string"/>
+            <assertNotEquals stepKey="assertNotEqualsTokenSecret" expected="($grabAccessTokenSecret)" actual="$grabReauthorizedAccessTokenSecret" actualType="string"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml]: Redundant indent.

File: app/code/Magento/Integration/Test/Mftf/ActionGroup/AdminAllowResourcesAccessIntegrationActionGroup.xml (hunk 2 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Integration/Test/Mftf/ActionGroup/AdminAllowResourcesAccessIntegrationActionGroup.xml]: Done

File: app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml (hunk 3 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml]: Done

File: app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml (hunk 4 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml]: Done

File: app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml (hunk 5 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml]: Done

File: app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml (hunk 6 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Integration/Test/Mftf/Test/AdminReAuthorizeTokensIntegrationEntityTest.xml]: Done

**Pull request merged**